Sources/Forum.php

```diff
@@ -336,14 +611,23 @@ protected function main(): array|string
 
 		// Check if the user should be disallowed access.
 		User::$me->kickIfBanned();
+	}
 
+	/**
+	 * The main forum loader.
+	 *
+	 * This method handles the main forum logic, such as checking permissions,
+	 * logging user activity, and tracking forum statistics.
+	 */
+	protected function main(): void
+	{
 		// If we are in a topic and don't have permission to approve it then duck out now.
 		if (!empty(Topic::$topic_id) && empty(Board::$info->cur_topic_approved) && !User::$me->allowedTo('approve_posts') && (User::$me->id != Board::$info->cur_topic_starter || User::$me->is_guest)) {
 			ErrorHandler::fatalLang('not_a_topic', false);
 		}
```

Comment:
Would this be able to be decoupled from here now?

Reply:
Perhaps, yeah. Where would you rather do this?

Reply:
Probably the equivalent in `Topic` of `Board::load()`

Reply:
Sounds good